<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Caviarbf Manual - Fine-mapping Documentation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="../css/highlight.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Caviarbf Manual";
    var mkdocs_page_input_path = "caviarbf_manual.md";
    var mkdocs_page_url = "/caviarbf_manual/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js"></script>
  <script src="../js/modernizr-2.8.3.min.js"></script>
  <script type="text/javascript" src="../js/highlight.pack.js"></script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Fine-mapping Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Introduction</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Data Preprocessing</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../preprocessing/file_format/">1. File formatting</a>
                </li>
                <li class="">
                    
    <a class="" href="../preprocessing/file_split/">2. File splitting</a>
                </li>
                <li class="">
                    
    <a class="" href="../preprocessing/ld_matrix_generation/">3. LD matrix generation</a>
                </li>
                <li class="">
                    
    <a class="" href="../preprocessing/functional_annotation_generation/">4. Functional annotation generation</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../software_running/">Software Running</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Caviarbf Manual</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#caviarbf">caviarbf</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#r-version">R version</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../reference/">Reference</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Fine-mapping Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Caviarbf Manual</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="caviarbf">caviarbf</h1>
<h2 id="r-version">R version</h2>
<p>Since the documentation of caviarbf is not clear enough and sometimes mixes the documentation of the C++ version and the R version this documentation serves as a supplement</p>
<h3 id="dependencies">Dependencies</h3>
<p>The R version depends on <code>Rcpp</code>, <code>RcppEigen</code> and <code>glmnet</code>, remember to install these package before installing caviarbf </p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>By 2018-06-28, there is an error when using the lastest version of <code>glmnet</code> (2.0-16). According to the <a href="https://bitbucket.org/Wenan/caviarbf/issues/2/error-running-test_packager">author's own explanation</a>, "This is a problem with the latest glmnet update. So far glmnet 2.0-10 seems to be working. I will try to see whether I can make it work with the latest glmnet (2.0-13)</p>
<p>When testing with <code>glmnet</code> (2.0-10), there is no error"</p>
</div>
<h3 id="software-detail">Software Detail</h3>
<p>Following information are copied from caviarbf R package help directly</p>
<p>Please type <code>?caviarbfFineMapping</code> in R for the usage of the interface.</p>
<h4 id="usage">Usage</h4>
<pre><code>caviarbfFineMapping(lociListFile, maxCausal, nSample, priorType01 = 0, 
priorValue = c(0.1, 0.2, 0.4), exact = T, eps = 0, useIdentityMatrix = F, 
BFFileListFile = NA, priorProb = 0, fittingMethod = "glmnetLASSOMin", 
hyperParamSelection = "cv", nFold = 5, K = 10, rThreshold = 0.2, 
pvalueThreshold = 0.05, lambda = c(2^seq(from = -15, by = 2, to = 5), 100, 1000, 10000, 1e+05, 1e+06), 
alpha = c(0, 0.2, 0.3, 0.5, 0.7, 0.8, 1), annotationIndices = -1, 
annotationSuffix = ".annotations", LDSuffix = ".LD", outputPrefix, keepBF = F, 
overwriteExistingResults = F, maxIter = 50, deltaLik = 0.01, 
useParallel = F, ncores = 1, verbose = F)
</code></pre>
<h4 id="argument">Argument</h4>
<ul>
<li>
<p><code>lociListFile</code> A file listing all the z score files.</p>
</li>
<li>
<p><code>maxCausal</code> The assumed maximal number of causal variants in each locus</p>
</li>
<li>
<p><code>nSample</code> The number of individuals used to calculate summary statistics</p>
</li>
<li>
<p><code>priorType01</code> 0 to specify sigmaa. An experimental 1 is used to specify the proportion of the phenotype variance explained (pve)</p>
</li>
</ul>
<hr />
<p>If <code>priorType01 = 0</code></p>
<ul>
<li><code>priorValue</code> If priorType01 is 0, it specifies the value of sigmaa. Otherwise, it is the pve value. For sigmaa, a vector of values can be used and the final Bayes factors are averaged. For GWAS fine mapping, c(0.1, 0.2, 0.4) might be a good option. For eQTL, c(0.1, 0.2, 0.4, 0.8, 1.6) might be a good option to accout for large eQTL effects.</li>
</ul>
<hr />
<ul>
<li>
<p><code>exact</code> Whether to calculate the exact Bayes factors. This is <strong>only</strong> valid for <em>quantitative traits</em>. For <em>binary traits</em>, <strong>only</strong> approximated Bayes factors are available</p>
</li>
<li>
<p><code>eps</code> The small value to add to the correlation matrix. This is useful when the LD matrix is from a reference panel</p>
</li>
<li>
<p><code>useIdentityMatrix</code> If true, use the identity matrix as the LD matrix. This improves the computing speed when only considering one causal variant for each locus</p>
</li>
<li>
<p><code>BFFileListFile</code> Specify the files of precomputed Bayes factors corresponding to each locus in the lociListFile. If NA, the Bayes factors will be calculated first. Bayes factors can precomputed for each locus using the C++ program This has the advantage of running in parallel, and can saved for later fine mapping.</p>
</li>
<li>
<p><code>priorProb</code> The initial probability of each variant being causal. If it is 0, the initial probability will be the number of loci divided by the number of all variants</p>
</li>
<li>
<p><code>hyperParamSelection</code> Method to select hyperparameter lambda and alpha. <code>"cv"</code> for loci based cross validation to select lambda and alpha, <code>"aic"</code>, <code>"bic"</code>, <code>"bic2"</code> uses AIC, BIC models. For the BIC model, <code>"bic2"</code> use the number of SNPs as the number of data points, <code>"bic"</code> uses the number of individuals as the number of data points, but they often show similar results. <code>"topK"</code> produces the top significant and relative independent annotations. No penalization is used in the fitting. As a by-product, it also outputs the pvalues for each individual annotation when considered separately</p>
</li>
</ul>
<hr />
<p>If <code>hyperParamSelection = "cv"</code></p>
<ul>
<li>
<p><code>fittingMethod</code> Specify different penalization models: <code>glmnetLASSOMin</code> for L1 penalization, <code>glmnetRidgeMin</code> for L2 penalization, <code>glmnetENETMin</code> for elastic net penalization. This is only used when <code>hyperParamSelection</code> is <code>"cv"</code>.</p>
</li>
<li>
<p><code>alpha</code> The alpha values for parameter selection, only used when the fitting method is <code>glmnetENETMin</code></p>
</li>
<li>
<p><code>nFold</code> The fold of cross validation, only used when <code>hyperParamSelection</code> is <code>"cv"</code>. If the number of loci is less than then nFold, the number of loci is used</p>
</li>
</ul>
<hr />
<p>If <code>hyperParamSelection = "topK"</code></p>
<ul>
<li><code>K</code> Specify the top K annotations to select, used for <code>"topK"</code></li>
<li><code>rThreshold</code> Specify the upper bound of the correlation among top annotations, used for <code>"topK"</code></li>
<li><code>pvalueThreshold</code> Specify the p value threshold for top annotations, used for <code>"topK"</code></li>
</ul>
<hr />
<ul>
<li>
<p><code>lambda</code> The lambda values for parameter selection. If lambda is 0, then no penalization is used in the fitting. This can be used for statistical testing, e.g., comparing two nested annotation models.</p>
</li>
<li>
<p><code>annotationIndices</code> A vector to specify annotations. If it is -1, use all annotations, 0 for only the intercept, otherwise it specifies the corresponding columns of annotations. Default uses all annotations.</p>
</li>
<li>
<p><code>annotationSuffix</code> The annotation suffix, default ".annotations"</p>
</li>
<li>
<p><code>LDSuffix</code> The correlation file suffix, default ".LD"</p>
</li>
<li>
<p><code>outputPrefix</code> The prefix of the output files</p>
</li>
<li>
<p><code>keepBF</code> Whether to keep produced Bayes factor files</p>
</li>
<li>
<p><code>overwriteExistingResults</code> Whether to overwrite existing output files</p>
</li>
<li>
<p><code>maxIter</code> The maximal iteration in the EM method</p>
</li>
<li>
<p><code>deltaLik</code> The minimum improvement of log likelihood to continue the EM iteration</p>
</li>
<li>
<p><code>useParallel</code> Whether to use library parallel to use multiple cores This can increase the speed of calculate PIPs in the EM iteration</p>
</li>
<li>
<p><code>ncores</code> Number of cores to use in parallel when <code>useParallel</code> is T</p>
</li>
<li>
<p><code>verbose</code> If true, more information are printed</p>
</li>
</ul>
<h4 id="value">value</h4>
<p>The results are saved in several files.  </p>
<p><code>&lt;outputPrefix&gt;.marginal</code> saves the indices of all variants following the order in the list file and the their PIPs sorted in a descring order.</p>
<p><code>&lt;outputPrefix&gt;.marginalz</code> includes the extra region index, the variant ID and the z score.</p>
<p><code>&lt;outputPrefix&gt;.loglik</code> stores the relative log likelihood.</p>
<p><code>&lt;outputPrefix&gt;.gamma</code> stores the estimated annotation effect size and the effect size of standardized annotation.</p>
<p><code>&lt;outputPrefix&gt;.log</code> stores the running log.</p>
<p><code>&lt;outputPrefix&gt;.time</code> saves the running time.</p>
<h4 id="note">Note</h4>
<p><code>"topK"</code> First, annotations are ranked by their individual increase in likelihood compared to the model without any annotations. Then the top k annotations are used</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../reference/" class="btn btn-neutral float-right" title="Reference">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../software_running/" class="btn btn-neutral" title="Software Running"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
      <p>Copyright © 2018 Hongcheng Yao</p>
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../software_running/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../reference/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js"></script>
      <script src="../search/require.js"></script>
      <script src="../search/search.js"></script>

</body>
</html>
